### PR TITLE
Winch Update Check

### LINKED ISSUE #7: Winch Update Check
While possibly more gimmick-y than genuinely useful, it would be neat to have a built-in check for a newer version of Winch which displays a little label in the main menu if an update is available.

The latest release tag on GitHub could be used for reference, as the repository tags follow the Winch version naming scheme and are therefore compatible with `VersionUtil.IsSameOrNewer()`

### PR DESCRIPTION
**Pull request by bdlm-dev**
Basic version check using Octokit.net, has worked fine in my testing but does require that the 1.2MB dll be included with Winch.
[Octokit.zip](https://github.com/Hacktix/Winch/files/11347281/Octokit.zip)
Resolves #7

### CODE REVIEW

File: Winch/Core/Initializer.cs [hunk 1]
```diff
@@ -48,5 +51,24 @@ private static void InitializeDevConsole()
             term.AddComponent<Terminal>();
             UnityEngine.Object.DontDestroyOnLoad(term);
         }
+
+        private static async void CheckForUpdate()
+        {
+            var client = new GitHubClient(new ProductHeaderValue("WinchFetchLatest"));
+            var releases = await client.Repository.Release.GetAll("Hacktix", "Winch");
+            var latest = releases[0];
+
```
Comment (bdlm-dev) [inline Winch/Core/Initializer.cs:60]: There is an 'unauthenticated rate limit' of I believe 60 requests/hour but that shouldn't be a problem.

**Comment by bdlm-dev** [comment]
On hindsight, would probably be good to have this toggleable in the config to avoid devs spamming their api.

**Comment by Ashiepaws** [comment]
Hmm, not exactly a huge fan of the extra dll just for update checks...
And yes, if there's a rate limit it should definitely be configurable.

**Comment by bdlm-dev** [comment]
Yeah i’m not sure it’s worth it given the extra dll, and wasn’t particularly able to find other options that wouldn’t also have a dependency.

**Comment by bdlm-dev** [comment]
Alright can't find a way to do it without dependencies, going to just close this for now.

**Comment by SloopyNoSleep** [comment]
Why not just make a simple request to `https://github.com/Hacktiz/Winch/releases/latest` and then check the re-direct link whether it has the latest or not?

Additionally it would be simpler if releases don't have the winch version in the zip name because then you can always link to `https://github.com/Hacktiz/Winch/releases/latest/Winch.zip` and have it resolve appropriately

**Pull request closed without merging**